decent-dao/fractal-interface
# [BUGFIX] Contracts Store in state not being updated

I'll explain some of the code changes in a review, But basically two things in this PR

## Fix contracts `asSigner` not properly being stored in state

As a DAO is loaded and stored in state in `useGovernanaceContracts`, the wallet provider may not be connect to a user's wallet yet and the contract's `asSigner` was being connected via a `FallbackProvider` and would not update when the user's connection status settles. 

This bug shows it self when attempting to vote and executing.

### The Fix
Remove storing Contracts in state. I thought of a few things I could ultimately do, but the best thing was to just save the retrieved address instead state (keeping the basic workflow the same) and as needed, attach the address to the appropriate contract.

## Testing

- Load up a Token DAO. (ERC20 or ER721). 
- Create a proposal
- Go to the proposal and Refresh the screen on that page
- Vote (should work just fine, I refresh a few times just to be sure no flakyness)
- Wait for voting period to end
- Before Executing, Refresh the browser. Then attempt to execute.


Refreshing on the details screen, when there is a actionable thing to do is best way to show this bug. Take same proposal to 
https://app.dev.fractalframework.xyz/ to see this bug.
